### PR TITLE
Change MGSmootherBlock to use shared memory pool

### LINKED ISSUE #392: Make MGSmootherBlock similar to MGSmoother-derived classes
The `MGSmoother`-derived classes had the constructor that takes a `VectorMemory` object as argument deprecated for a long time already, in favor of a version that uses an internally provided memory object. (And I'm about to remove these deprecated functions in a second.)

On the other hand, `MGSmootherBlock` did not make this change, i.e. the constructor of this class still requires such an object and there are no alternatives. It should make the same change.

### PR DESCRIPTION
**Pull request by class4kayaker**
Old constructor has been deprecated, and the test (`multigrid/smoother_block`) has been modified to use the new constructor.

Should fix #392.

I would appreciate another set of eyes for style, etc as the change was made partially by referencing diffs from far back in the history, and therefore I may have missed some critical elements. Also, I was uncertain whether there was a reason for `MGSmootherBlock` to not be based on `MGSmoother` other than history. I decided to derive from `MGSmoother`, but if there is a reason this should not be done, I would appreciate the correction.

### CODE REVIEW

File: include/deal.II/multigrid/mg_block_smoother.h [hunk 1]
```diff
@@ -125,35 +122,14 @@ class MGSmootherBlock
...
    * Reverse?
    */
   bool reverse;
 
   /**
    * Memory for auxiliary vectors.
    */
-  VectorMemory<BlockVector<number> > &mem;
-
+  SmartPointer<VectorMemory<BlockVector<number>>, MGSmootherBlock<MatrixType, RelaxationType, number >> mem;
```
Comment (bangerth) [inline include/deal.II/multigrid/mg_block_smoother.h]: You'll need a space between `>>` for C++98.

**Comment by kronbichler** [comment]
Could you also please add a changelog entry in the specific improvements section?

Apart from that, this change looks good to me. Since I've not used this class myself and therefore might miss something, I will wait with merging for a day or two.

**Comment by kronbichler** [comment]
Could you please squash your commits into two commits, one for the code and test changes and one for the changelog?

**Comment by kronbichler** [comment]
Great!

**Pull request merged**